src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -613,6 +613,121 @@ A list of all generated pages. It is an object with one property.
 
 - `pathname` - the finalized path of the page.
 
+### Defining custom hooks
+
+As an integration author, you can define custom hooks that your integration will call. This allows for runtime interactions between your integration and other integrations. Other integrations can provide implementations for your custom hooks and your integration can call those at the appropriate moment in your integration's lifecycle.
+
+#### Defining hook types
+
+Provide the type for your custom hook by augmenting Astro’s `IntegrationHooks` interface within your integration:
+
+```ts title="types.ts"
+// on an implementation file
```

Comment:
The file is a `something.ts` not a `something.d.ts`. Meaning this is part of actual code that you can add wherever you want not just an ambient declaration (TS terminology) that has to go on a special file.

But yeah, it is already in the title

Reply:
Gotcha. If you think it's helpful information to include, to differentiate from a `*.d.ts` file, then maybe we could add it to the text above? Something like

> Provide the type for your custom hook by augmenting Astro’s `IntegrationHooks` interface in any implementation file within your integration.

or ideally, if there's maybe a general way to describe the file that someone would use this in, then this could be a nice way to introduce it

> Provide the type for your custom hook by augmenting Astro’s `IntegrationHooks` interface in YOUR integration's [ADJECTIVE] file:

Reply:
OK, sounds good!